solidity/test/token_stake/TestTokenStake.js

```diff
@@ -267,52 +273,68 @@ contract('TokenStaking', function(accounts) {
 
   describe("undelegate", async () => {
     it("should let operator undelegate", async () => {
-      await delegate(operatorOne, stakingAmount)
+      let tx = await delegate(operatorOne, stakingAmount)
+      let createdAt = (await web3.eth.getBlock(tx.receipt.blockNumber)).timestamp
 
-      await mineBlocks(initializationPeriod)
+      await increaseTimeTo(createdAt + initializationPeriod + 1)
       await stakingContract.undelegate(operatorOne, {from: operatorOne})
       // ok, no revert
     })
 
     it("should not allow third party to undelegate", async () => {
-      await delegate(operatorOne, stakingAmount)
+      let tx = await delegate(operatorOne, stakingAmount)
+      let createdAt = (await web3.eth.getBlock(tx.receipt.blockNumber)).timestamp
 
-      await mineBlocks(initializationPeriod)
+      await increaseTimeTo(createdAt + initializationPeriod + 1)
       await expectThrowWithMessage(
         stakingContract.undelegate(operatorOne, {from: operatorTwo}),
         "Only operator or the owner of the stake can undelegate"
       )
     })
 
-    it("should permit undelegating at the block when initialization " + 
+    it("should permit undelegating at the time when initialization " + 
     "period passed", async () => {
-      await delegate(operatorOne, stakingAmount)
+      let tx = await delegate(operatorOne, stakingAmount)
+      let createdAt = (await web3.eth.getBlock(tx.receipt.blockNumber)).timestamp
 
-      await mineBlocks(initializationPeriod)
+      await increaseTimeTo(createdAt + initializationPeriod + 1)
       await stakingContract.undelegate(operatorOne, {from: operatorOne})
       // ok, no revert
     })
 
-    it("should not permit undelegating at the block before initialization " + 
+    it("should not permit undelegating at the time before initialization " + 
     "period passed", async () => {
-      await delegate(operatorOne, stakingAmount)
+      let tx = await delegate(operatorOne, stakingAmount)
+      let createdAt = (await web3.eth.getBlock(tx.receipt.blockNumber)).timestamp
+
+      await increaseTimeTo(createdAt + initializationPeriod - 1)
```

Comment:
[This](https://github.com/keep-network/keep-core/pull/1512/files/d339004f82283c04495c8638676b7e87b08fe6c2#r397889897) comment still applies, I think we should have `await increaseTimeTo(createdAt + initializationPeriod)` here.